server/routes/users.js

```diff
@@ -116,10 +116,35 @@ export default (storage, scriptManager) => {
 
             return {
               user,
-              memberships: [ ]
+              memberships: []
             };
           });
       })
+      .then(data => {
+        if (!data.user.identities) {
+          data.connection = {};
+          return data;
+        }
+
+        const identities = data.user.identities.filter(identity => identity.provider === 'auth0');
+        const name = identities[0] && identities[0].connection;
+
+        if (!name) {
+          data.connection = {};
+          return data;
+        }
+
+        return req.auth0.connections.getAll({ name, fields: 'id' })
+          .then(connection => req.auth0.connections.get({ id: connection[0].id, fields: 'enabled_clients' }))
```

Comment:
Is there any danger in assuming that a connection will be returned?
that is, could `connection[0]` be undefined, throwing an error for `connection[0].id`

Reply:
where/how often is "get single user" called?

Reply:
We're taking connection name from the user object, then searching for its id. As long as api2 is ok, there shouldn't be any problems.

It is being called every time, when somebody is opening user's details.

Reply:
> As long as api2 is ok, there shouldn't be any problems
That's a big assumption. I think we should put in a better guard.